x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/percentile.tsx

```diff
@@ -0,0 +1,186 @@
+  type: 'percentile',
+  displayName: i18n.translate('xpack.lens.indexPattern.percentile', {
+    defaultMessage: 'Percentile',
+  }),
+  input: 'field',
+  getPossibleOperationForField: ({ aggregationRestrictions, aggregatable, type: fieldType }) => {
+    if (fieldType === 'number' && aggregatable && !aggregationRestrictions) {
+      return {
+        dataType: 'number',
+        isBucketed: false,
+        scale: 'ratio',
+      };
+    }
+  },
+  isTransferable: (column, newIndexPattern) => {
+    const newField = newIndexPattern.getFieldByName(column.sourceField);
+
+    return Boolean(
+      newField &&
+        newField.type === 'number' &&
+        newField.aggregatable &&
+        !newField.aggregationRestrictions
+    );
+  },
+  getDefaultLabel: (column, indexPattern, columns) =>
+    ofName(getSafeName(column.sourceField, indexPattern), column.params.percentile),
+  buildColumn: ({ field, previousColumn, indexPattern }) => {
+    const existingFormat =
+      previousColumn?.params && 'format' in previousColumn?.params
+        ? previousColumn?.params?.format
+        : undefined;
+    const existingPercentileParam =
+      previousColumn?.operationType === 'percentile' && previousColumn?.params.percentile;
+    const newPercentileParam = existingPercentileParam || DEFAULT_PERCENTILE_VALUE;
+    return {
+      label: ofName(getSafeName(field.name, indexPattern), newPercentileParam),
+      dataType: 'number',
+      operationType: 'percentile',
+      sourceField: field.name,
+      isBucketed: false,
+      scale: 'ratio',
+      params: {
+        format: existingFormat,
+        percentile: newPercentileParam,
+      },
+    };
+  },
+  onFieldChange: (oldColumn, field) => {
+    return {
+      ...oldColumn,
+      label: ofName(field.displayName, oldColumn.params.percentile),
+      sourceField: field.name,
+    };
+  },
+  toEsAggsFn: (column, columnId, _indexPattern) => {
+    return buildExpressionFunction<AggFunctionsMapping['aggPercentiles']>('aggPercentiles', {
+      id: columnId,
+      enabled: true,
+      schema: 'metric',
+      field: column.sourceField,
+      percents: [column.params.percentile],
+    }).toAst();
+  },
+  getEsAggsSuffix: (column) => {
+    const value = column.params.percentile;
+    return `.${value}`;
+  },
+  getErrorMessage: (layer, columnId, indexPattern) =>
+    getInvalidFieldMessage(layer.columns[columnId] as FieldBasedIndexPatternColumn, indexPattern),
+  paramEditor: function PercentileParamEditor({
+    layer,
+    updateLayer,
+    currentColumn,
+    columnId,
+    indexPattern,
+  }) {
+    const [inputValue, setInputValue] = useState(String(currentColumn.params.percentile));
+
+    const inputValueAsNumber = Number(inputValue);
+    // an input is value if it's not an empty string, parses to a valid number, is between 0 and 100 (inclusive)
+    // and is an integer
+    const inputValueIsValid = isValidNumber(inputValue, true, 99, 1);
+
+    useDebounce(
```

Comment:
Mind that this `useDebounce` implementation triggers two times at bootstrap: there's the `useDebounceWithOptions` version which can skip the first render